cli/models.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+
+# Proxyctrl - API backed CLI for the proxy container
+# version 0.1
+# This program aims to allow for the ease of use of the proxy via terminal interface
+# Author - Amos Jones
+# Project Supervisor - William Mantly
+# 12/19/2019
+
+# import python3 libraries
+import requests
+
+#class proxyctrl handles calls to the api
+class model:
+	# Starting here, these functions are internal to the class and should not be accessed from outside 
+	def __init__(self, base_url=None, token=None):
+		self.base_url = base_url or 'http://localhost:8300/'
+		self.token = token or ''
+		
+		self.headers = {
+			'auth-token'  : self.token
+		}
+		
+	def __process_error(self, res):
+		if res.status_code in range(200, 299):
+			return None
+		
+		if res.status_code == 401:
+			# do stuff for bad login
+			pass
+		return res.reason
+
+	def __build_headers(self):
```

Comment:
Consider having `__build_headers()` return the headers, like is done here; https://gist.github.com/wmantly/26ed89e34f2a0af562cf107db2b2136d#file-godaddy-py-L17